### PR TITLE
Preserve packed attribute in C translated struct

### PR DESCRIPTION
**Pull request by via**
Solves #4084

### CODE REVIEW

**Comment by andrewrk** [comment]
Nice work - see https://github.com/ziglang/zig/blob/master/CONTRIBUTING.md#improving-translate-c (towards the end) for how to add test cases for this.

File: src/zig_clang.cpp [hunk 1]
```diff
@@ -1592,6 +1592,14 @@ const char* ZigClangVarDecl_getSectionAttribute(const struct ZigClangVarDecl *se
     return nullptr;
 }
 
+bool ZigClangRecordDecl_isPacked(const ZigClangRecordDecl *zig_record_decl) {
```
Comment (daurnimator) [inline src/zig_clang.cpp]: Should this be a more general get attribute?

Comment (LemonBoy) [reply src/zig_clang.cpp]: My only nit is that it should be named `getPackedAttribute` to match the other getters.

File: test/run_translated_c.zig [hunk 2]
```diff
@@ -23,4 +23,17 @@ pub fn addCases(cases: *tests.RunTranslatedCContext) void {
...
+
+    cases.add("struct initializer - packed",
+        \\#define _NO_CRT_STDIO_INLINE 1
+        \\#include <stdio.h>
+        \\#include <stdint.h>
+        \\struct s {uint8_t x,y;
+        \\          uint32_t z;} __attribute__((packed)) s0 = {1, 2};
+        \\int main() {
+        \\  /* sizeof nor offsetof currently supported */
+        \\  printf("%x\n", (intptr_t)&s0.z - (intptr_t)&s0.x);
```
Comment (LemonBoy) [inline test/run_translated_c.zig]: You can sidestep any problem with varargs with something like:

```suggestion
        \\  if (((intptr_t)&s0.z - (intptr_t)&s0.x) != 2) abort();
```

Comment (andrewrk) [reply test/run_translated_c.zig]: Yes, and in fact, would be preferred, when not testing `printf` directly. We've got the "hello world" test for `printf`.

**Comment by andrewrk** [comment]
Nice work.

**Pull request merged**